core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -3435,6 +3437,43 @@ object SparkContext extends Logging {
     supplement(DRIVER_JAVA_OPTIONS)
     supplement(EXECUTOR_JAVA_OPTIONS)
   }
+
+  private def supplementJavaGCOptions(conf: SparkConf): Unit = {
+    val javaFlagCommand = "java -XX:+PrintFlagsFinal -version"
```

Comment:
I think this actually does not guarantee the same way how the current JVM is created .. Different `java` exec could be used.

Reply:
The new implementation checks if a GC algorithm is specified in the configuration: https://github.com/apache/spark/pull/51796/commits/38eacf02d8e9c1fddcfa2b3f1566e0e6e85673d4